presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -176,6 +177,12 @@ public ActualProperties visitApply(ApplyNode node, List<ActualProperties> inputP
         }
 
         @Override
+        public ActualProperties visitLateralJoin(LateralJoinNode node, List<ActualProperties> inputProperties)
+        {
+            return inputProperties.get(0); // lateral node input (outer query)
```

Comment:
I don't think so, but I might be wrong. I don't feel much confident around this code.